apps/authentication/backends/saml2/views.py

```diff
@@ -146,12 +151,21 @@ def get_sp_settings(self):
                 },
                 'singleLogoutService': {
                     'url': f"{sp_host}{reverse('authentication:saml2:saml2-logout')}"
-                }
+                },
             }
         }
         sp_settings['sp'].update(attrs)
         advanced_settings = self.get_advanced_settings()
         sp_settings.update(advanced_settings)
+
+        # load sp cert and key for encryption if advanced_settings.security.authnRequestsSigned is True
+        if advanced_settings['authnRequestsSigned']:
```

Comment:
Is the parameter authnRequestsSigned more appropriate in security

Reply:
ops, using `get` method here would increase robustness, will update this PR.

Reply:
Should the parameter format be like this: advanced_settings = {' security ': {' authnRequestsSigned' : True}}

Reply:
1. Yap, the key point here is to set  appropriate values for  `x509cert` and `privakeyKey` in sp_settings['sp']  which is needed for python-saml library to sign the SAML request,  which is now not supported since the certs and keys are simply not loaded and hence will trigger an `can not load key` error.

2. The `authnRequestsSigned` field  is used by  `python-saml` library to tell if to sign the SAML request or not.

3.  It is recommended to add an prompt on web console to set the values for `authnRequestsSigned` for IdPs may require signature provided, and when not, the SAML Response would be `Signature Required` and lead to unsuccessful SSO Login.